# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: netwerk/dns/nsDNSService2.cpp
# Commit: f778b37ce90f
# Full Hash: f778b37ce90fcc2171a280cd56d6472e7aa62177
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2024-07-03 16:15:19
# Description:
#   Bug 1903719 - Make InitMockHTTPSRecord return null when type is not RESOLVE_TYPE_HTTPSSVC, r=sunil
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D214548
# ==============================================================================

diff -r 6733585b9ee6 -r f778b37ce90f netwerk/dns/nsDNSService2.cpp
--- a/netwerk/dns/nsDNSService2.cpp	Wed Jul 03 09:16:26 2024 +0000
+++ b/netwerk/dns/nsDNSService2.cpp	Wed Jul 03 09:32:28 2024 +0000
@@ -500,6 +500,11 @@
   if (NS_SUCCEEDED(status) ||
       mFlags & nsIDNSService::RESOLVE_WANT_RECORD_ON_ERROR) {
     MOZ_ASSERT(hostRecord, "no host record");
+    if (!hostRecord) {
+      mListener->OnLookupComplete(this, nullptr, NS_ERROR_UNKNOWN_HOST);
+      mListener = nullptr;
+      return;
+    }
     if (hostRecord->type != nsDNSService::RESOLVE_TYPE_DEFAULT) {
       rec = new nsDNSByTypeRecord(hostRecord);
     } else {